stock-price-points/database/index.js

```diff
@@ -1,23 +1,147 @@
+  {name: 'id'},
+  {name: 'divindex'},
+  {name: 'divaverage'},
+  {name: 'divstockspurchased'}
+  ], {table: 'stockdistribution'});
+
+function getCompany(companyAbbriev, callback) {
+  let queryString = `SELECT * FROM stockprices INNER JOIN stockdistribution 
+    ON stockprices.distributionid = stockdistribution.id 
+    WHERE stockprices.companyabbriev = \'${companyAbbriev}\'`;
+  db.result(queryString)
+    .then(data => { 
+      if (data.rows.length === 0) { callback(null, false); }
+      else { callback(null, data.rows); }
+    })
+    .catch(err => { callback(err); })
+}
+
+function addCompany(companyEntry, distributionEntries, callback) {
+  let { companyabbriev, company, stockspurchased, 
+    yearhigh, yearlow, yearavg, currentprice } = companyEntry;
+  let queryString = `INSERT INTO stockprices (companyabbriev, company, stockspurchased, 
+    yearhigh, yearlow, yearavg, currentprice, distributionid) 
+    VALUES ('${companyabbriev}', '${company}', ${stockspurchased}, 
+    ${yearhigh}, ${yearlow}, ${yearavg}, ${currentprice}, DEFAULT) RETURNING distributionid;`;
+  db.one(queryString)
+    .then(result => {
+        // console.log(`DistId: ${result.distributionid}`);
+        // console.log(`${companyEntry.companyabbriev} inserted`);
+        
+        let distributionEntriesWithId = distributionEntries.map(div => { 
+          return {...div, id: result.distributionid}; 
+        })
+        let distributionInsert = pgp.helpers.insert(distributionEntriesWithId, distributionColSet);
+        db.none(distributionInsert)
+          .then(() => { 
+            console.log(`Distribution inserted`); 
+            callback();
+          })
+          .catch((err) => callback(err) )
+      })
+    .catch((err) => { 
+      // console.log('Error inserting company');
+      callback(err); });   
+}
+
+function updateCompany(companyEntry, distributionEntries, callback) {
+  let companyUpdate = pgp.helpers.update(companyEntry, ['company', 
+    'stockspurchased','yearhigh', 'yearlow', 'yearavg', 'currentprice'], 'stockprices') + 
+    `WHERE companyabbriev = '${companyEntry.companyabbriev}' RETURNING distributionid`;
+  db.one(companyUpdate)
+    .then(result => {
+      // console.log(`DistId: ${result.distributionid}`);
+      // console.log(`${companyEntry.companyabbriev} updated`);
+      let distributionUpdate = pgp.helpers.update(distributionEntries, 
+        ['?divindex', 'divaverage', 'divstockspurchased'], 'stockdistribution') + 
+        `WHERE v.divindex = t.divindex AND t.id = ${result.distributionid}`; 
+      db.none(distributionUpdate)
+        .then(() => { 
+          // console.log(`Distribution updated`); 
+          callback();
+        })
+        .catch((err) => callback(err) )
+    })
+    .catch(err => { callback(err); })
+}
+
+function deleteCompany(companyAbbriev, callback) {
+  let queryString = `SELECT distributionid FROM stockprices 
+    WHERE companyabbriev = \'${companyAbbriev}\';`;
+  db.one(queryString)
```

Comment:
rather than using a callback, if you return the promise, you can do a .then/.catch to the place you called the function